### PR TITLE
Remove goto statements in aws_iot_shadow_parser.c

### PR DESCRIPTION
**Pull request by qiutongs**
*Issue #, if available:* https://sim.amazon.com/issues/TS-10929

*Description of changes:*
Remove goto statements in aws_iot_shadow_parser.c

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/aws/shadow/src/aws_iot_shadow_parser.c [hunk 1]
```diff
@@ -144,52 +141,53 @@ AwsIotShadowError_t _AwsIotShadow_ParseErrorDocument( const char * pErrorDocumen
...
-                                   &pMessage,
-                                   &messageLength ) == true )
+                                   &codeLength ) == true )
     {
-        IotLogWarn( "Code %lu: %.*s.",
-                    code,
-                    messageLength,
-                    pMessage );
+        /* Code must be in error document. */
+        AwsIotShadow_Assert( ( pCode > pErrorDocument ) &&
```
Comment (aggarw13) [inline libraries/aws/shadow/src/aws_iot_shadow_parser.c]: (Minor) Not sure if this assert check is really necessary. The first expression checks if the code is present after the starting address of the error document. Do you know why that is a requirement? If this function is carrying the knowledge of the JSON encoding format, and is thus, checking that the binary error code's location exists after the starting location, I'm not a fan of this parser to carry serialization format level information.

(I understand that this is out of the scope of the PR, but I would like it to be removed if not required)

File: libraries/aws/shadow/src/aws_iot_shadow_parser.c [hunk 2]
```diff
@@ -144,52 +141,53 @@ AwsIotShadowError_t _AwsIotShadow_ParseErrorDocument( const char * pErrorDocumen
...
+                             ( pCode + codeLength < pErrorDocument + errorDocumentLength ) );
+
+        /* Convert the code to an unsigned integer value. */
+        code = ( uint32_t ) strtoul( pCode, NULL, 10 );
```
Comment (aggarw13) [inline libraries/aws/shadow/src/aws_iot_shadow_parser.c]: Not sure if the string to integer conversion is actually required here, as it seems like the error code will only be logged, which can be done with the string version stored in `pCode`

Comment (qiutongs) [reply libraries/aws/shadow/src/aws_iot_shadow_parser.c]: That code is also used here:
```
status = _codeToShadowStatus( code );
```

File: libraries/aws/shadow/src/aws_iot_shadow_parser.c (hunk 1 above)
Comment (qiutongs) [reply libraries/aws/shadow/src/aws_iot_shadow_parser.c]: I agree with you on this. @gordonwang0 Do you think so?

Comment (qiutongs) [reply libraries/aws/shadow/src/aws_iot_shadow_parser.c]: @leegeth Can you comment on this?

**Pull request merged**